### PR TITLE
Various usability improvements to experiment submission

### LINKED ISSUE #152: Error messages hide compatibility info message
Steps to reproduce:

1. Run an experiment with an incompatible model and protocol combination
2. Quickly click on the matrix again, to see the message that they're not compatible
3. Go back to the matrix view
4. Click again, now you see an error message, and the original information is gone

I liked the original information! And if there's multiple output / multiple errors we need to see them all

### PR DESCRIPTION
**Pull request by jonc125**
This may address some of your issues @MichaelClerx ; I think it fixes #152 at least. Perhaps also part of #149? It also makes the notes more useful, which may impact #147.

It also fixes https://rollbar.com/jonc125/Web-Lab/items/37.

### CODE REVIEW

File: weblab/experiments/processing.py [hunk 1]
```diff
@@ -56,12 +57,18 @@ def submit_experiment(model, model_version, protocol, protocol_version, user, re
...
-        )
+        try:
+            version, created = ExperimentVersion.objects.get_or_create(
+                experiment=experiment,
+                defaults={
+                    'author': user,
+                }
+            )
+        except MultipleObjectsReturned:
+            print('Multi objects')
```
Comment (MichaelClerx) [inline weblab/experiments/processing.py]: Are we printing rather than logging in the front-end?

Comment (jonc125) [reply weblab/experiments/processing.py]: Oops, that was debug output left in by mistake!

File: weblab/experiments/tests/test_views.py [hunk 2]
```diff
@@ -660,6 +660,32 @@ def test_submits_experiment(
         assert data['newExperiment']['versionId'] == version.id
         assert data['newExperiment']['expName'] == version.experiment.name
 
+        # Check re-submit also works if there are multiple versions already
+        exp_version2 = recipes.experiment_version.make(
+            status='SUCCESS',
+            experiment=version.experiment,
+        )
+        print(version, exp_version2)
```
Comment (MichaelClerx) [inline weblab/experiments/tests/test_views.py]: One more :D

**Comment by MichaelClerx** [comment]
Looks alright to me, although I'm not sure how to test this in combination with the fitting branch?

**Comment by jonc125** [comment]
> I'm not sure how to test this in combination with the fitting branch?

You can merge this branch into the fitting branch.

**Comment by MichaelClerx** [comment]
Happy to merge this into master now

**Pull request merged**